Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp

```diff
@@ -110,34 +110,49 @@ void WebUserContentControllerProxy::addProcess(WebProcessProxy& webProcessProxy)
         webProcessProxy.addMessageReceiver(Messages::WebUserContentControllerProxy::messageReceiverName(), identifier(), *this);
 }
 
+#if 0
+struct UserContentControllerParameters {
+    UserContentControllerIdentifier identifier;
+    Vector<std::pair<ContentWorldIdentifier, String>> userContentWorlds;
+    Vector<WebUserScriptData> userScripts;
+    Vector<WebUserStyleSheetData> userStyleSheets;
+    Vector<WebScriptMessageHandlerData> messageHandlers;
+#if ENABLE(CONTENT_EXTENSIONS)
+    Vector<std::pair<WebCompiledContentRuleListData, URL>> contentRuleLists;
+#endif
+};
+#endif
+
 UserContentControllerParameters WebUserContentControllerProxy::parameters() const
 {
-    UserContentControllerParameters parameters;
-
-    parameters.identifier = identifier();
-    
-    ASSERT(parameters.userContentWorlds.isEmpty());
-    parameters.userContentWorlds = WTF::map(m_associatedContentWorlds, [](auto& identifier) {
+    auto userContentWorlds = WTF::map(m_associatedContentWorlds, [](auto& identifier) {
         auto* world = API::ContentWorld::worldForIdentifier(identifier);
         RELEASE_ASSERT(world);
         return world->worldData();
     });
 
+    Vector<WebUserScriptData> userScripts;
     for (auto userScript : m_userScripts->elementsOfType<API::UserScript>())
-        parameters.userScripts.append({ userScript->identifier(), userScript->contentWorld().identifier(), userScript->userScript() });
+        userScripts.append({ userScript->identifier(), userScript->contentWorld().identifier(), userScript->userScript() });
 
+    Vector<WebUserStyleSheetData> userStyleSheets;
     for (auto userStyleSheet : m_userStyleSheets->elementsOfType<API::UserStyleSheet>())
-        parameters.userStyleSheets.append({ userStyleSheet->identifier(), userStyleSheet->contentWorld().identifier(), userStyleSheet->userStyleSheet() });
+        userStyleSheets.append({ userStyleSheet->identifier(), userStyleSheet->contentWorld().identifier(), userStyleSheet->userStyleSheet() });
 
-    parameters.messageHandlers = WTF::map(m_scriptMessageHandlers, [](auto entry) {
+    auto messageHandlers = WTF::map(m_scriptMessageHandlers, [](auto entry) {
         return WebScriptMessageHandlerData { entry.value->identifier(), entry.value->world().identifier(), entry.value->name() };
     });
 
+    return {
+        identifier()
+        , WTFMove(userContentWorlds)
```

Comment:
```suggestion
        WTFMove(userContentWorlds),
```

Reply:
Your proposal doesn't play well with the `#if ENABLE(CONTENT_EXTENSIONS)` below.